CMakeLists.txt

```diff
@@ -607,17 +607,19 @@ if(EL_TESTS)
       set_source_files_properties("${DRIVER}" PROPERTIES
         OBJECT_DEPENDS "${PREPARED_HEADERS}")
       target_link_libraries(tests-${TYPE}-${TESTNAME} El)
+      set(TEST_OUTPUT_NAME ${TESTNAME})
+      if(NOT NO_BINARY_SUBDIRECTORIES)
```

Comment:
I think this would be easier to read if it was of the form:
```
if(NO_BINARY_SUBDIRECTORIES)
  set(TEST_OUTPUT_NAME ${TESTNAME})
  set(TEST_OUTPUT_DIR "")
else()
  ...
endif()
```
and if we added a short note on the CMake option `NO_BINARY_SUBDIRECTORIES` at the top of the file.